core/src/main/java/org/apache/accumulo/core/clientImpl/bulk/BulkImport.java

```diff
@@ -548,22 +548,22 @@ public SortedMap<KeyExtent,Bulk.Files> computeFileToTabletMappings(FileSystem fs
         context.instanceOperations().getSystemConfiguration(), tableProps, tableId);
 
     for (FileStatus fileStatus : files) {
-      Path filePath = fileStatus.getPath();
+      UnassignedTabletFile file = UnassignedTabletFile.of(fs, fileStatus.getPath());
       CompletableFuture<Map<KeyExtent,Bulk.FileInfo>> future = CompletableFuture.supplyAsync(() -> {
         try {
           long t1 = System.currentTimeMillis();
           List<KeyExtent> extents =
-              findOverlappingTablets(context, extentCache, filePath, fs, fileLensCache, cs);
+              findOverlappingTablets(context, extentCache, file, fs, fileLensCache, cs);
           // make sure file isn't going to too many tablets
-          checkTabletCount(maxTablets, extents.size(), filePath.toString());
-          Map<KeyExtent,Long> estSizes = estimateSizes(context.getConfiguration(), filePath,
+          checkTabletCount(maxTablets, extents.size(), file.toString());
+          Map<KeyExtent,Long> estSizes = estimateSizes(context.getConfiguration(), file,
               fileStatus.getLen(), extents, fs, fileLensCache, cs);
           Map<KeyExtent,Bulk.FileInfo> pathLocations = new HashMap<>();
           for (KeyExtent ke : extents) {
-            pathLocations.put(ke, new Bulk.FileInfo(filePath, estSizes.getOrDefault(ke, 0L)));
+            pathLocations.put(ke, new Bulk.FileInfo(file.getPath(), estSizes.getOrDefault(ke, 0L)));
```

Comment:
I think we might want a follow on issue to fix BulkImport in general, maybe the same time we update the file length cache to use a better key than a String. Strings are used extensively in BulkImport throughout the entire API so we probably should try and make it better wherever we can and use TabletFile references while not breaking serialization. I didn't make BulkImport a priority here because BulkImport is importing new files of course and wouldn't really need to worry about fencing. 

I'm wondering if I can refactor classes like [LoadPlan](https://github.com/apache/accumulo/blob/4d7f81954be05e414f814aa116ed65536edf7ba6/core/src/main/java/org/apache/accumulo/core/data/LoadPlan.java#L93) to use a TabletFile instead of a String as part of this but don't know if making breaking changes to the constructor would be ok or not for compatibility as I'm not sure if it's public or where else it's used as I haven't looked much into Bulk Import before.